wicket-core/src/test/java/org/apache/wicket/markup/MarkupParserTest.java

```diff
@@ -133,12 +133,12 @@ final void test1() throws Exception
 		log.info("tok(4)=" + tokens.get(4));
 		log.info("tok(5)=" + tokens.get(5));
 
-		assertTrue(tokens.get(0).equals("This is a test "));
+		assertEquals("This is a test ", tokens.get(0));
```

Comment:
`tokens.get(0)` is of type `RawMarkup` https://github.com/apache/wicket/blob/master/wicket-core/src/main/java/org/apache/wicket/markup/RawMarkup.java

so `tokens.get(0).equals(_some_string_)` works due to `equals` implementation

this assert need to be rewritten as 
`assertEquals("This is a test ", tokens.get(0).toCharSequence());`
OR
`assertEquals("This is a test ", tokens.get(0).toString());`

also other 2 asserts with `tokens.get()`